### PR TITLE
Feature filter remove_keys

### PR DESCRIPTION
**Pull request by vbotka**
##### SUMMARY

ansible.utils.remove_keys does not work properly. Fixing is not efficient.
Feature request #8442

##### ISSUE TYPE

- New Module/Plugin Pull Request

##### COMPONENT NAME

community.general.remove_keys

##### ADDITIONAL INFORMATION

See:  community.general.keep_keys #8438 #8456

### CODE REVIEW

File: plugins/filter/remove_keys.py [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+
+from ansible.errors import AnsibleFilterError
+from ansible.module_utils.six import string_types
+from ansible.module_utils.common._collections_compat import Mapping, Sequence
+
+import re
+
+
+def remove_keys(data, target=None, matching_parameter='equal'):
+    """remove specific keys from dictionaries in a list"""
```
Comment (felixfontein) [inline plugins/filter/remove_keys.py:86]: The code here is very similar to the one in #8439. How about moving the common parts of the implementation to `plugins/plugin_utils/_keys_filter.py`, and only handle the differences in the filter plugin code?

**Review by felixfontein — COMMENTED** [review]
Similar comments as in #8439, and the extra one below :)

File: plugins/filter/remove_keys.py (hunk 1 above)
Comment (vbotka) [reply plugins/filter/remove_keys.py:86]: Thank you! I agree. I'll create a new PR.

**Comment by vbotka** [comment]
IMO, this PR is ready too. Would it be possible to review it? A diff shows the trivial changes

```bash
shell> diff keep_keys.py remove_keys.py
```

File: plugins/filter/remove_keys.py [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+      _input:
+        description:
+          - A list of dictionaries.
+          - Top level keys must be strings.
+        type: list
+        elements: dictionary
+        required: true
+      target:
+        description:
+          - A single key or key pattern to keep, or a list of keys or keys patterns to keep.
```
Comment (felixfontein) [inline plugins/filter/remove_keys.py:28]:
```suggestion
          - A single key or key pattern to remove, or a list of keys or keys patterns to remove.
```

File: plugins/filter/remove_keys.py [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+
+  # The results of above examples 6-9 are all the same.
+  r:
+    - {k1_x1: B0, k2_x2: [C0], k3_x3: foo}
+    - {k1_x1: B1, k2_x2: [C1], k3_x3: bar}
+'''
+
+RETURN = '''
+  _value:
+    description: The list of dictionaries with selected keys.
```
Comment (felixfontein) [inline plugins/filter/remove_keys.py:99]:
```suggestion
    description: The list of dictionaries with selected keys removed.
```

**Review by felixfontein — COMMENTED** [review]
I found two docs changes, otherwise it looks good!

**Comment by vbotka** [comment]
Done. Thank you!

**Review by felixfontein — APPROVED** [review]

**Comment by felixfontein** [comment]
@vbotka thanks for your contribution!

**Comment by vbotka** [comment]
Thank you!

**Pull request merged**